src/screens/DiscoveryScreenV2/fragment/UsersFragment.js

```diff
@@ -280,38 +265,69 @@ const UsersFragment = ({
     });
   };
 
-  const renderUsersItem = () => {
-    const initFollowingUsers = [];
-    const initUnfollowingUsers = [];
-
+  const initFollowingUsers = React.useMemo(() => {
+    const initialFollowingUsers = [];
     users.forEach((item) => {
       if (item.user?.user_id_follower || item.user_id_follower) {
-        initFollowingUsers.push(item);
-      } else {
-        initUnfollowingUsers.push(item);
+        initialFollowingUsers.push(item);
+      }
+    });
+
+    return initialFollowingUsers;
+  }, [users]);
+
+  const initUnfollowingUsers = React.useMemo(() => {
+    const initialUnfollowingUsers = [];
+    users.forEach((item) => {
+      if (!item.user?.user_id_follower && !item.user_id_follower) {
+        initialUnfollowingUsers.push(item);
       }
     });
 
-    const data = isFirstTimeOpen
-      ? withoutRecent
-        ? initialUsers.length !== 0
-          ? initialUsers
-          : [{separator: true}, ...initUnfollowingUsers]
-        : [{separator: true}, ...initUnfollowingUsers]
-      : unfollowedUsers.length !== 0
-      ? [{separator: true}, ...unfollowedUsers]
-      : [];
-
-    const firstData = isFirstTimeOpen
-      ? withoutRecent
-        ? initialUsers.length !== 0
-          ? initialUsers
-          : [...initFollowingUsers]
-        : [...initFollowingUsers]
-      : unfollowedUsers.length !== 0
-      ? [...followedUsers]
-      : followedUsers;
+    return initialUnfollowingUsers;
+  }, [users]);
+
+  const peopleYouMightKnowData = React.useMemo(() => {
+    if (!isFirstTimeOpen && unfollowedUsers.length === 0) {
+      return [];
+    }
+
+    if (!isFirstTimeOpen && unfollowedUsers) {
+      return [{separator: true}, ...unfollowedUsers];
+    }
+
+    if (!withoutRecent) {
+      return [{separator: true}, ...initUnfollowingUsers];
+    }
+
+    if (initialUsers.length === 0) {
+      return [{separator: true}, ...initUnfollowingUsers];
+    }
+
+    return initialUsers;
+  }, [users, unfollowedUsers]);
+
+  const initialAccordionData = React.useMemo(() => {
+    if (!isFirstTimeOpen && unfollowedUsers.length !== 0) {
+      return [...followedUsers];
+    }
+
+    if (!isFirstTimeOpen) {
+      return followedUsers;
+    }
 
+    if (!withoutRecent) {
+      return [...initFollowingUsers];
+    }
+
+    if (initialUsers.length !== 0) {
+      return initialUsers;
+    }
+
+    return [...initFollowingUsers];
+  }, [users, unfollowedUsers, initialUsers]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `useMemo` for calculating `initFollowingUsers`, `initUnfollowingUsers`, `peopleYouMightKnowData`, and `initialAccordionData` is a good practice for performance optimization. It ensures that these computations are only re-run when the `users` or `unfollowedUsers` change, rather than on every render.



<!-- This is an auto-generated comment by OSS CodeRabbit -->